nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-processors/src/main/java/org/apache/nifi/processors/elasticsearch/QueryElasticsearchHttp.java

```diff
@@ -0,0 +1,410 @@
+
+            if (flowFile != null) {
+                session.remove(flowFile);
+            }
+        } catch (IOException ioe) {
+            logger.error(
+                    "Failed to read from Elasticsearch due to {}, this may indicate an error in configuration "
+                            + "(hosts, username/password, etc.). Routing to retry",
+                    new Object[] { ioe.getLocalizedMessage() }, ioe);
+            if (flowFile != null) {
+                session.transfer(flowFile, REL_RETRY);
+            }
+            context.yield();
+
+        } catch (RetryableException e) {
+            logger.error(e.getMessage(), new Object[] { e.getLocalizedMessage() }, e);
+            if (flowFile != null) {
+                session.transfer(flowFile, REL_RETRY);
+            }
+            context.yield();
+        } catch (Exception e) {
+            logger.error("Failed to read {} from Elasticsearch due to {}", new Object[] { flowFile,
+                    e.getLocalizedMessage() }, e);
+            if (flowFile != null) {
+                session.transfer(flowFile, REL_FAILURE);
+            }
+            context.yield();
+        }
+    }
+
+    private int getPage(final Response getResponse, final URL url, final ProcessContext context,
+            final ProcessSession session, FlowFile flowFile, final ComponentLog logger,
+            final long startNanos, boolean targetIsContent)
+            throws IOException {
+        List<FlowFile> page = new ArrayList<>();
+        final int statusCode = getResponse.code();
+
+        if (isSuccess(statusCode)) {
+            ResponseBody body = getResponse.body();
+            final byte[] bodyBytes = body.bytes();
+            JsonNode responseJson = parseJsonResponse(new ByteArrayInputStream(bodyBytes));
+            JsonNode hits = responseJson.get("hits").get("hits");
+
+            for(int i = 0; i < hits.size(); i++) {
+                JsonNode hit = hits.get(i);
+                String retrievedId = hit.get("_id").asText();
+                String retrievedIndex = hit.get("_index").asText();
+                String retrievedType = hit.get("_type").asText();
+
+                FlowFile documentFlowFile = null;
+                if (flowFile != null) {
+                    documentFlowFile = targetIsContent ? session.create(flowFile) : session.clone(flowFile);
+                } else {
+                    documentFlowFile = session.create();
+                }
+
+                JsonNode source = hit.get("_source");
+                documentFlowFile = session.putAttribute(documentFlowFile, "es.index", retrievedIndex);
+                documentFlowFile = session.putAttribute(documentFlowFile, "es.type", retrievedType);
+
+                if (targetIsContent) {
+                    documentFlowFile = session.putAttribute(documentFlowFile, "filename", retrievedId);
+                    documentFlowFile = session.write(documentFlowFile, out -> {
```

Comment:
Since _source is a JSON object, I recommend setting the mime.type attribute to application/json when the target is content.